cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -389,6 +392,39 @@ private static File hijackYarnConfFile(File confFile) {
   }
 
   /**
+   * Check that the file is a mapred-site.xml file, and return a temp copy of it to which are added
+   * necessary options. If it is not a mapred-site.xml file, return it as is.
+   */
+  private static File hijackMapredConfFile(File confFile) {
+    Configuration conf = new Configuration(false);
+    try {
+      conf.addResource(confFile.toURI().toURL());
+    } catch (MalformedURLException e) {
+      LOG.error("File {} is malformed.", confFile, e);
+      throw Throwables.propagate(e);
+    }
+
+    String mrAppClassPath = conf.get(MRJobConfig.MAPREDUCE_APPLICATION_CLASSPATH,
+                                       MRJobConfig.DEFAULT_MAPREDUCE_APPLICATION_CLASSPATH);
+
+    // Add the pwd/* at the beginning of classpath. Without this change, old jars from mr framework classpath
+    // get into classpath.
+    mrAppClassPath = "$PWD/*," + mrAppClassPath;
+
+    conf.set(MRJobConfig.MAPREDUCE_APPLICATION_CLASSPATH, mrAppClassPath);
+
+    File newMapredConfFile = new File(Files.createTempDir(), "mapred-site.xml");
```

Comment:
In that case, use the temp directory inside the data directory so that when master starts again, it will gets cleanup. See https://github.com/caskdata/cdap/blob/develop/cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkRuntimeService.java#L120